tests/test_arrow_dataset.py

```diff
@@ -3119,7 +3119,7 @@ def test_pickle_dataset_after_transforming_the_table(in_memory, method_and_param
 
 
 @pytest.mark.skipif(
-    os.name == "nt" and (os.getenv("CIRCLECI") == "true" or os.getenv("GITHUB_ACTIONS") == "true"),
+    os.name in ["nt", "posix"] and (os.getenv("CIRCLECI") == "true" or os.getenv("GITHUB_ACTIONS") == "true"),
```

Comment:
I don't think you need this ? The test currently passes on `main` on linux

Reply:
It doesn't on Python 3.7. Notice the `test_dummy_dataset_serialize_s3` [failure](https://github.com/huggingface/datasets/runs/7467455443?check_suite_focus=true) in one of the previous test

Reply:
Ok. This test is almost never run then, Maybe let's remove it completely in another PR, and make another one to make sure S3 export works as expected. Maybe we can use the `mockfs` fixture defined in https://github.com/huggingface/datasets/pull/4724 instead

Reply:
This issue seems related to https://github.com/spulec/moto/issues/4750, but neither bumping `moto` to `3.0.0` (CI runs forever) nor pinning responses to `0.16` (the serialization fails) helps, so I agree we can disable these tests for now and replace them in another PR with the `mockfs` fixture